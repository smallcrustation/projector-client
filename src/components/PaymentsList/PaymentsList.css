.PaymentsList{
  margin: 15px 0 50px 0;
  border: 1px solid black;
}

.PaymentsList section{
  display: flex;
  flex-direction: column;
}

.PaymentsList p{
  text-align: center;
}

.PaymentsList__btn{
  padding: 20px;
}

#PaymentsList__table{
  margin: 0 auto;
  margin-bottom: 20px;
  line-height: 1.6;
  border-collapse: collapse;
}