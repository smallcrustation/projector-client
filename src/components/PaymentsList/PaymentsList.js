import React from 'react'
import accounting from 'accounting'

import { ProjectContextConsumer } from '../../contexts/ProjectContext'

import './PaymentsList.css'

export default class PaymentsList extends React.Component {
  renderPayments() {
    return (
      <ProjectContextConsumer>
        {value => {
          console.log(value.payments)
          if(!value.payments) return <p>Loading...</p>
          if(value.payments.length<1) return <p>No Payments</p>

          return value.payments.map((payment,i) => {
            console.log(payment)
            return (
              <li key={payment.id}>
                Payment {payment.payment_number}:{' '}
                {accounting.formatMoney(payment.total_amount)}
                {/* (pending) is a quick fix for the last payment, later needs to toggle  */}
                {value.payments.length-1===i?'(pending)':''}
              </li>
            )
          })
        }}
      </ProjectContextConsumer>
    )
  }

  render() {

    let content = this.renderPayments()

    return (
      <div className="PaymentsList">
        <h3>Payments</h3>
        <section>
          <ul>
            {content}
          </ul>
          <button onClick={this.props.onClickNewPaymentRequest}>
            New Payment Request
          </button>
        </section>
      </div>
    )
  }
}
