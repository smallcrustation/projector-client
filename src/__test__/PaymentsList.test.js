import React from 'react'
import ReactDOM from 'react-dom'

import PaymentsList from '../components/PaymentsList/PaymentsList'
import renderer from 'react-test-renderer'

describe('App Testing', () => {
  it('Renders without crashing', () => {
    const div = document.createElement('div')

    ReactDOM.render(<PaymentsList />, div)

    ReactDOM.unmountComponentAtNode(div)
  })

  it('Renders with no crashing and matches snapshot', () => {
    const tree = renderer.create(<PaymentsList />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
