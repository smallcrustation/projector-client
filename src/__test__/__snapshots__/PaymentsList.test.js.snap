// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Testing Renders with no crashing and matches snapshot 1`] = `
<div
  className="PaymentsList"
>
  <h3>
    Payments
  </h3>
  <section>
    <p>
      No Context
    </p>
    <button
      className="PaymentsList__btn"
    >
      New Payment Request
    </button>
  </section>
</div>
`;
